### ISSUE TITLE
Export Critical Log fails

### ISSUE THREAD

**Issue by marionbarker** [LoopKit/Loop#2081]
**Describe the bug**
Not able to export critical logs from test phone. Edited because this is related to building main after building dev on the phone. Going from main to dev is OK. Going from dev to main triggers this error, even after restoring to dev.

**Issue Report**

[Loop Report 2023-10-02 10:28:50-07:00.md](https://github.com/LoopKit/Loop/files/12785540/Loop.Report.2023-10-02.10.28.50-07.00.md)

**Screenshot of failure**
![not-able-to-export](https://github.com/LoopKit/Loop/assets/19607791/e366d78e-eb71-4de6-883f-d1ca47480248)

**Comment by marionbarker** [comment]
I'm wondering if this is a "test" phone problem.

* On my "real" phone, running iOS 16.7, built with commit a7cc83c, export critical logs work fine and look nominal when I airdrop to my computer.
* Repeated on "test" phone SE, running iOS 17.0.2. also built with commit a7cc83c, 
   * I get the critical log error. I had the phone attached to xcode and captured the graphic, below, on the failure, with text copied as well.
* Then tried on "test" phone 8, running iOS 16.6, also built with commit a7cc83c and had same error.

In case this is a consequence of building lots of different versions as dev evolved, and switching back and forth between main and dev, on the test phones, I deleted app (but left health data alone).
* iPhone 8, fresh app - no issues with export
* only the current day has content but that is as expected

The SE phone still has the app on it in case there are tests to be tried to fix this without deleting the app.
* Updated the SE to iOS 17.0.3 - no change.

```
Failure during critical event log export: keyNotFound(CodingKeys(stringValue: "value", intValue: nil), Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "carbsOnBoard", intValue: nil)], debugDescription: "No value associated with key CodingKeys(stringValue: \"value\", intValue: nil) (\"value\").", underlyingError: nil))
```

![critical-log-failure-ios-17](https://github.com/LoopKit/Loop/assets/19607791/cb514434-d7be-4dc2-a1ec-50b4eea0ed4c)

**Comment by marionbarker** [comment]
Not sure if I can reproduce this. Keep it open for now.

**Comment by marionbarker** [comment]
Bump

**Comment by marionbarker** [comment]
I just reproduced this on a test phone.
* I have been using this phone, SE running iOS 17.2 with latest dev
* Exported critical logs - no issue
* Installed Loop 3.2.3 (from TestFlight) on this phone and cannot export critical logs
* Reinstalled dev (from TestFlight) and cannot export critical logs
* Reinstalled same build of dev from Xcode and cannot export critical logs

**Comment by marionbarker** [comment]
I just had to delete the app from my test phone again because changing between dev and main means I can no longer export critical event logs and I needed to confirm that both dev and main report that carbs are edited or deleted in the Carbs.json file.

I did confirm both main and dev report that information.

This is not an issue for test phones, but if we have "real" users switching to dev and then back to main, they could be in a situation where their critical logs cannot be exported. Returning to dev does not solve the problem, the error remains.

**Comment by marionbarker** [comment]
Bump

**Comment by marionbarker** [comment]
This is still a problem.

Test:
* start with phone running dev - export critical logs works
* build main over dev - export critical logs fails
* build dev over main (immediately) - export critical logs works
* build main over dev - confirm export critical logs fails again
    * do a manual bolus while in main
    * return to dev - export critical logs fails

**Comment by marionbarker** [comment]
Fixed in Loop 3.4.
